### PR TITLE
Encryption at rest.

### PR DESCRIPTION
**Pull request by marinsalinas**
```json
{
    "AWSTemplateFormatVersion" : "2010-09-09",
    "Description" : "This template creates an encryption at rest on the MongoDB Atlas API, this will be billed to your Atlas account.",
    "Parameters" : {
      "ProjectName" : {
        "Type" : "String",
        "Description" : "",
        "ConstraintDescription" : ""
      },
      "AccessKeyID" : {
        "Type" : "String",
        "Description" : "",
        "ConstraintDescription" : ""
      },
      "SecretAccessKey" : {
        "Type" : "String",
        "Description" : "",
        "ConstraintDescription" : ""
      },
      "CustomerMasterKeyID" : {
        "Type" : "String",
        "Description" : "",
        "ConstraintDescription" : ""
      },
      "Region" : {
        "Type" : "String",
        "Description" : "",
        "ConstraintDescription" : ""
      }
    },
    "Mappings" : {},
    "Resources" : {
      "Project": {
        "Type": "MongoDB::Atlas::Project",
        "Properties": {
          "Name": { "Ref" : "ProjectName" },
          "Orgid": "{{resolve:secretsmanager:mongodb/atlas/secrets:SecretString:AtlasMongoDBOrgID}}",
          "ApiKeys": {
            "PublicKey": "{{resolve:secretsmanager:mongodb/atlas/secrets:SecretString:AtlasMongoDBPublicKey}}",
            "PrivateKey": "{{resolve:secretsmanager:mongodb/atlas/secrets:SecretString:AtlasMongoDBPrivateKey}}"
          }
        }
      },
      "EncryptionAtRest": {
        "Type": "MongoDB::Atlas::EncryptionAtRest",
        "Properties": {
          "ProjectId": { "Ref" : "Project" },
          "AwsKms": {
            "AccessKeyID": { "Ref" : "AccessKeyID" },
            "SecretAccessKey": { "Ref" : "SecretAccessKey" },
            "CustomerMasterKeyID": { "Ref" : "CustomerMasterKeyID" },
            "Enabled": true,
            "Region": { "Ref" : "Region" }
          },
          "ApiKeys": {
            "PublicKey": "{{resolve:secretsmanager:mongodb/atlas/secrets:SecretString:AtlasMongoDBPublicKey}}",
            "PrivateKey": "{{resolve:secretsmanager:mongodb/atlas/secrets:SecretString:AtlasMongoDBPrivateKey}}"
          }
        }
      }
    },
    "Outputs" : {
      "ProjectId" : {
        "Description:" : "Id of the project",
        "Value" : { "Ref" : "Project" }
      },
      "EncryptionId" : {
        "Description:" : "Id of the encryption at rest",
        "Value" : { "Ref" : "EncryptionAtRest" }
      }
    }
  }
```

### CODE REVIEW

**Comment by themantissa** [comment]
Added Chris on as a reviewer for when done.

File: encryption-at-rest/.gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# macOS
+.DS_Store
```
Comment (thetonymaster) [inline encryption-at-rest/.gitignore]: Remove this file, we have one at root.

File: encryption-at-rest/go.sum [hunk 2]
```diff
@@ -0,0 +1,47 @@
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/Sectorbob/mlab-ns2 v0.0.0-20171030222938-d3aa0c295a8a h1:KFHLI4QGttB0i7M3qOkAo8Zn/GSsxwwCnInFqBaYtkM=
+github.com/Sectorbob/mlab-ns2 v0.0.0-20171030222938-d3aa0c295a8a/go.mod h1:D73UAuEPckrDorYZdtlCu2ySOLuPB5W4rhIkmmc/XbI=
```
Comment (thetonymaster) [inline encryption-at-rest/go.sum]: remove this file, we have one at root.

File: encryption-at-rest/go.mod [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module github.com/mongodb/mongodbatlas-cloudformation-resources/encryption-at-rest
+
+go 1.13
+
+require (
+	github.com/avast/retry-go v2.4.3+incompatible // indirect
```
Comment (thetonymaster) [inline encryption-at-rest/go.mod]: remove this file, we have one at root.

File: encryption-at-rest/mongodb-atlas-encryptionatrest.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        "ProjectId"
+    ],
+    "readOnlyProperties": [""],
+    "primaryIdentifier": [
+        "/properties/ProjectId"
+    ],
+    "handlers": {
+        "create": {
+            "permissions": [
+                "initech:CreateReport"
```
Comment (thetonymaster) [inline encryption-at-rest/mongodb-atlas-encryptionatrest.json]: remove all the `initech:` permissions and just leave an empty string

**Comment by marinsalinas** [comment]
Hello @thetonymaster, is this tested?

File: encryption-at-rest/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+Next steps:
+
+1. Populate the JSON schema describing your resource, `mongodb-atlas-encryptionatrest.json`
+2. The RPDK will automatically generate the correct resource model from the
+   schema whenever the project is built via Make.
+   You can also do this manually with the following command: `cfn-cli generate`
+3. Implement your resource handlers by adding code to provision your resources in your resource handler's methods.
+
+Please don't modify files `model.go and main.go`, as they will be automatically overwritten.
```
Comment (shum) [inline encryption-at-rest/README.md]:
```suggestion
Please don't modify files `model.go` and `main.go`, as they will be automatically overwritten.
```

File: encryption-at-rest/cmd/resource/model.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package resource
+
+import "github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+
+/*
+This file is autogenerated, do not edit;
+changes will be undone by the next 'generate' command.
+
+Updates to this type are made my editing the schema file
```
Comment (shum) [inline encryption-at-rest/cmd/resource/model.go]:
```suggestion
Updates to this type are made by editing the schema file
```

File: encryption-at-rest/cmd/resource/resource.go [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+			Enabled:             currentModel.AwsKms.Enabled.Value(),
+			AccessKeyID:         *currentModel.AwsKms.AccessKeyID.Value(),
+			SecretAccessKey:     *currentModel.AwsKms.SecretAccessKey.Value(),
+			CustomerMasterKeyID: *currentModel.AwsKms.CustomerMasterKeyID.Value(),
+			Region:              *currentModel.AwsKms.Region.Value(),
+		},
+		GroupID: *currentModel.ProjectId.Value(),
+	}
+
+	log.Printf("Arguments: Project ID: %s, Request %#+v", projectID, encryptionAtRest)
```
Comment (shum) [inline encryption-at-rest/cmd/resource/resource.go]: leftover?

File: encryption-at-rest/cmd/resource/resource.go [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+	_, _, err = client.EncryptionsAtRest.Create(context.Background(), encryptionAtRest)
+
```
Comment (shum) [inline encryption-at-rest/cmd/resource/resource.go]: delete nl

File: encryption-at-rest/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+   schema whenever the project is built via Make.
+   You can also do this manually with the following command: `cfn-cli generate`
+3. Implement your resource handlers by adding code to provision your resources in your resource handler's methods.
+
+Please don't modify files `model.go` and `main.go`, as they will be automatically overwritten.
+
+## Attributes
+`ProjectId` *(required)* : The unique identifier of the project.<br>
+`AwsKms.AccessKeyID` *(required)* : The IAM access key ID with permissions to access the customer master key specified by customerMasterKeyID.<br>
+`AwsKms.CustomerMasterKeyID` *(required)* : he AWS customer master key used to encrypt and decrypt the MongoDB master keys.<br>
```
Comment (shum) [inline encryption-at-rest/README.md]:
```suggestion
`AwsKms.CustomerMasterKeyID` *(required)* : The AWS customer master key used to encrypt and decrypt the MongoDB master keys.<br>
```

File: encryption-at-rest/README.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+`AwsKms.Enabled` *(required)* : he IAM secret access key with permissions to access the customer master key specified by customerMasterKeyID.<br>
```
Comment (shum) [inline encryption-at-rest/README.md]:
```suggestion
`AwsKms.Enabled` *(required)* : The IAM secret access key with permissions to access the customer master key specified by customerMasterKeyID.<br>
```

File: encryption-at-rest/README.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+`AwsKms.SecretAccessKey` *(required)* : Specifies whether Encryption at Rest is enabled for an Atlas project. To disable Encryption at Rest, pass only this parameter with a value of false. When you disable Encryption at Rest, Atlas also removes the configuration details.<br>
+`AwsKms.Region` *(required)* : The AWS region in which the AWS customer master key exists.<br>
+`ApiKeys` *(required)* : The private and public keys of the MongoDB Atlas.<br>
```
Comment (shum) [inline encryption-at-rest/README.md]:
```suggestion
`ApiKeys` *(required)* : The private and public keys of the MongoDB Atlas project.<br>
```

Comment (shum) [reply encryption-at-rest/README.md]: Moderately sure this is what you mean?

**Pull request merged**